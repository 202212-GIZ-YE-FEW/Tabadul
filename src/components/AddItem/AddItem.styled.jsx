import styled from "@emotion/styled";
import Image from "next/image";

export const Pagecontainer = styled.div`
    margin: 0 0 40px 0;
`;
export const ItemHeader = styled.h2`
    font-weight: 700;
    font-size: 36px;
    color: #3c4347;
    text-align: center;
`;
export const Formcontainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
`;

export const FormItem = styled.form``;
export const ItemInput = styled.div`
    padding: 20px;
    margin-top: 10px;
`;

export const ItemImageInput = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 10px;
    margin: 10px 0 30px 0;
    width: 100%;
`;
export const ImagesDiv = styled.div`
    width: 100%;
    display: grid;
    grid-gap: 30px 10px;
    justify-content: center;
    place-items: center;
    grid-template-columns: repeat(auto-fit, minmax(100px, 400px));
`;
export const ItemImage = styled(Image)`
    border-radius: 20px;
    width: 100%;
    height: 300px;
    /* object-position: center top; */
    object-fit: ${(props) =>
        props.alt === "thumbnail" ? "cover" : " contain"};
`;
export const ItemTitle = styled.label`
    font-weight: 400;
    font-size: 24px;
    color: #3c4347;
    display: flex;
    align-items: flex-end;
    margin-left: 7px;
    padding: 10px;
    @media screen and (max-width: 820px) {
        font-size: 20px;
    }
`;
export const TitleInput = styled.input`
    font-family: inherit;
    border: none;
    width: 800px;
    height: 60.78px;
    top: 58.22px;
    background: #fdfdfd;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    font-size: 20px;
    padding-left: 20px;
    &::placeholder {
        font-weight: 400;
        font-size: 20px;
        color: #bbbbbb;
    }

    &:focus {
        outline: none !important;
        border: 2px solid #33956d;
    }
    @media screen and (max-width: 768px) {
        width: 348px;
        height: 50px;
    }
`;

export const ItemCatogry = styled.label`
    font-weight: 400;
    font-size: 24px;
    color: #3c4347;
    display: flex;
    align-items: flex-end;
    margin-left: 7px;
    padding: 10px;
    @media screen and (max-width: 768px) {
        font-size: 20px;
    }
`;

export const CatogryInput = styled.select`
    border: none;
    width: 800px;
    height: 60.78px;
    top: 58.22px;
    background: #fdfdfd;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    font-weight: 400;
    font-size: 20px;
    font-color: black;
    line-height: 30px;
    display: flex;
    align-items: center;
    padding: 10px 10px 10px 20px;
    color: #bbbbbb;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background-image: url("./images/Arrow.svg");
    background-repeat: no-repeat;
    background-position: right 20px center;
    @media screen and (max-width: 768px) {
        width: 348px;
        height: 50px;
    }
    &:focus {
        outline: none !important;
        border: 2px solid #33956d;
    }
`;
export const Selectoption = styled.option``;

export const ItemLocation = styled.label`
    font-weight: 400;
    font-size: 24px;
    color: #3c4347;
    display: flex;
    align-items: flex-end;
    margin-left: 7px;
    padding: 10px;
    @media screen and (max-width: 768px) {
        font-size: 20px;
    }
`;

export const LocationInput = styled.select`
    border: none;
    width: 800px;
    height: 60.78px;
    top: 58.22px;
    background: #fdfdfd;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    font-weight: 400;
    font-size: 20px;
    line-height: 30px;
    display: flex;
    align-items: center;
    padding: 10px 10px 10px 20px;
    color: #bbbbbb;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    background-image: url("./images/Arrow.svg");
    background-repeat: no-repeat;
    background-position: right 20px center;
    @media screen and (max-width: 768px) {
        width: 348px;
        height: 50px;
    }
    &:focus {
        outline: none !important;
        border: 2px solid #33956d;
    }
`;

export const Locationoption = styled.option``;

export const ItemDescription = styled.label`
    font-weight: 400;
    font-size: 24px;
    color: #3c4347;
    display: flex;
    align-items: flex-end;
    margin-left: 7px;
    padding: 10px;
    @media screen and (max-width: 768px) {
        font-size: 20px;
    }
`;
export const DescriptionInput = styled.textarea`
    font-family: inherit;
    border: none;
    width: 800px;
    height: 242.06px;
    top: 60.94px;
    background: #ffffff;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    font-size: 20px;
    padding-left: 20px;
    padding-top: 15px;
    line-height: 30px;
    &::placeholder {
        font-weight: 400;
        font-size: 20px;
        display: flex;
        align-items: center;
        color: #bbbbbb;
        /* padding-left: 20px;
        padding-top: 10px; */
    }
    &:focus {
        outline: none !important;
        border: 2px solid #33956d;
    }
    @media screen and (max-width: 768px) {
        width: 348px;
        height: 151px;
    }
`;

export const Itemupload = styled.label`
    font-weight: 400;
    font-size: 24px;
    color: #3c4347;
    display: flex;
    align-items: flex-end;
    margin-left: 20px;
    padding: 10px;
    @media screen and (max-width: 768px) {
        font-size: 20px;
    }
`;
export const UploadContainer = styled.div`
    display: flex;
    flex-direction: column;
    border: none;
    width: 380px;
    height: 100%;
    background: #ffffff;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    padding: 10px;
    position: relative;
    cursor: pointer;

    @media screen and (max-width: 768px) {
        width: 348px;
        height: 100%;
    }
`;

export const Uploadspan = styled.span`
    font-weight: 400;
    font-size: 20px;
    color: #bbbbbb;
    overflow-wrap: break-word;
    width: 100%;
`;

export const UploadInput = styled.input`
    width: 100%;
    height: 100%;
    opacity: 0;
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    /* border: 2px solid black; */
`;

export const Buttoncontainer = styled.div`
    display: flex;
    justify-content: space-evenly;
`;

export const ConfirmButton = styled.button`
    background-color: ${(props) => (props.imagesList ? "#33956d" : "grey")};
    color: ${(props) => (props.imagesList ? "white" : "black")};
    /* background: #33956d; */
    border-radius: 54px;
    border: none;
    width: 150px;
    height: 48.77px;
    text-align: center;
    text-color: #ffffff;
    @media screen and (max-width: 768px) {
        width: 117px;
        height: 38;
    }
`;
export const Confirmspan = styled.span`
    font-weight: 700;
    font-size: 24px;
    text-align: center;
    color: #ffffff;
    @media screen and (max-width: 768px) {
        font-size: 16px;
    }
`;

export const CancelButton = styled.button`
    border: 3px solid #33956d;
    border-radius: 54px;
    width: 150px;
    height: 48.77px;
    background-color: #ffffff;
    @media screen and (max-width: 768px) {
        width: 117px;
        height: 38;
    }
`;
export const Cancelspan = styled.span`
    font-weight: 700;
    font-size: 24px;
    text-align: center;
    color: #33956d;
    @media screen and (max-width: 768px) {
        font-size: 16px;
    }
`;

export const DisplayContainer = styled.div`
    display: flex;
    flex-direction: column;
    border: none;
    width: 380px;
    height: 100%;
    background: #ffffff;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    padding: 10px;
    position: relative;

    @media screen and (max-width: 768px) {
        width: 348px;
        height: 100%;
    }
`;

export const ErrorLabel = styled.label`
    color: #f07167;
    display: flex;
    align-items: center;
    margin-right: 10px;
    font-size: 17px;
`;
export const LabelDiv = styled.div`
    display: flex;
    justify-content: space-between;
`;
export const PlusDiv = styled.div`
    width: 93%;
    height: 400px;
    box-shadow: 1px 3px 9px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 40px;
    color: #6f6f6f;
    position: relative;
`;
export const DeleteIcon = styled.div`
    position: absolute;
    right: 10px;
    top: 15px;
    cursor: pointer;
`;
