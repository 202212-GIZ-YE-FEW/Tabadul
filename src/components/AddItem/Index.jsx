import { useTranslation, withTranslation } from "next-i18next";
import React, { useEffect, useState } from "react";

import {
    Buttoncontainer,
    CancelButton,
    Cancelspan,
    CatogryInput,
    ConfirmButton,
    Confirmspan,
    DeleteIcon,
    DescriptionInput,
    DisplayContainer,
    ErrorLabel,
    Formcontainer,
    FormItem,
    ImagesDiv,
    ItemCatogry,
    ItemDescription,
    ItemHeader,
    ItemImage,
    ItemImageInput,
    ItemInput,
    ItemLocation,
    ItemTitle,
    Itemupload,
    LabelDiv,
    LocationInput,
    Locationoption,
    Pagecontainer,
    PlusDiv,
    Selectoption,
    TitleInput,
    UploadContainer,
    UploadInput,
    Uploadspan,
} from "./AddItem.styled";
import { addItem, auth } from "@/utils/firebase";
import {
    deleteObject,
    getDownloadURL,
    getMetadata,
    ref,
    uploadBytesResumable,
} from "firebase/storage";
import { storage } from "@/utils/firebase";
import Link from "next/link";
import Router from "next/router";
import Image from "next/image";
import thumbnail from "../../../public/thumbnails/default-image.png";

function AddItem({ categoriesList, locationList }) {
    const { t } = useTranslation("addItem");
    function translateMe(word) {
        return t(word);
    }

    const [imagesList, setImagesList] = useState([]);
    const [errorMsg, setErrorMsg] = useState({
        title: "",
        location: "",
        category: "",
        description: "",
        image: "",
    });
    const current = new Date();

    function currentDate() {
        const date = current.getDate().toString().padStart(2, "0");
        const month = (current.getMonth() + 1).toString().padStart(2, "0");
        const year = current.getFullYear();
        return `${date}/${month}/${year}`;
    }

    const [item, setItem] = useState({
        title: "",
        description: "",
        category: "",
        location: "",
        image: "",
    });
    function addto(e) {
        e.preventDefault();
        const formatedDate = currentDate();
        const updateItem = {
            ...item,
            time: current.toLocaleTimeString(),
            date: formatedDate,
            image: imagesList,
            userId: auth?.currentUser?.uid,
        };
        let flagIsFilled = validateOnSubmit(updateItem);
        if (flagIsFilled === Object.entries(updateItem).length) {
            addItem(updateItem);
            alert("product has been added!");
            setTimeout(() => {
                Router.push("/Products");
            }, 1000);
        }
    }

    function validateOnSubmit(updateItem) {
        let flag = 0;
        Object.entries(updateItem).forEach(([key, value]) => {
            if (value.length === 0) {
                setErrorMsg((prev) => ({ ...prev, [key]: true }));
            } else {
                setErrorMsg((prev) => ({ ...prev, [key]: false }));
                flag++;
            }
        });
        return flag;
    }

    function validateOnChange(e) {
        const name = e.target.name;
        if (e.target.value.length <= 0) {
            setErrorMsg({ ...errorMsg, [name]: true });
        } else {
            setErrorMsg({ ...errorMsg, [name]: false });
        }
    }
    function handleChange(e) {
        validateOnChange(e);
        const name = e.target.name;
        const value = e.target.value;
        setItem({ ...item, [name]: value });
    }

    function handleSelect(e) {
        validateOnChange(e);
        const name = e.target.name;
        const index = e.target.selectedIndex;
        const el = e.target.childNodes[index];
        const option = el.getAttribute("id");

        setItem({
            ...item,
            [name]: { ...JSON.parse(e.target.value) },
        });
    }

    function handleImageFile(e) {
        validateOnChange(e);
        uploadImage(e.target.files);
    }

    function uploadImage(imgs) {
        Array.from(imgs).map((img) => {
            const storageRef = ref(storage, `/files/${img.name}`);
            const uploadTask = uploadBytesResumable(storageRef, img);

            uploadTask.on(
                "state_changed",
                (snapshot) => {
                    const percent = Math.round(
                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
                    );
                },
                (err) => console.log(err),
                () => {
                    // download url
                    getDownloadURL(uploadTask.snapshot.ref).then(
                        async (url) => {
                            // console.log(url.name);
                            let imageName = await getMetadata(storageRef).then(
                                (metadata) => {
                                    return metadata.name;
                                }
                            );
                            console.log(imageName);
                            setImagesList((prev) => [
                                ...prev,
                                { url: url, name: imageName },
                            ]);
                        }
                    );
                }
            );
        });
    }

    function deleteImage(imageName) {
        const desertRef = ref(storage, `/files/${imageName}`);

        deleteObject(desertRef)
            .then(() => {
                const updatedList = imagesList.filter((img) => {
                    return img.name !== imageName;
                });
                setImagesList(updatedList);
            })
            .catch((error) => {
                console.log(error);
            });
    }

    useEffect(() => {}, [errorMsg]);

    return (
        <Pagecontainer>
            <ItemHeader>{translateMe("AddItem")} </ItemHeader>
            <Formcontainer>
                <FormItem method='post' onSubmit={addto}>
                    <ItemInput>
                        <LabelDiv>
                            <ItemTitle>{translateMe("Title")}</ItemTitle>
                            {errorMsg.title && (
                                <ErrorLabel errorMsg={errorMsg}>
                                    {translateMe("TitleisRequired")}
                                </ErrorLabel>
                            )}
                        </LabelDiv>
                        <TitleInput
                            placeholder={translateMe("titlePlaceholder")}
                            type='text'
                            name='title'
                            onChange={handleChange}
                        />
                    </ItemInput>
                    <ItemInput>
                        <LabelDiv>
                            <ItemCatogry>{translateMe("Catogry")}</ItemCatogry>
                            {errorMsg.category && (
                                <ErrorLabel errorMsg={errorMsg}>
                                    {translateMe("CategoryisRequired")}
                                </ErrorLabel>
                            )}
                        </LabelDiv>
                        <CatogryInput
                            placeholder={translateMe(
                                "Selectcatogryplaceholder"
                            )}
                            type='dropdown'
                            name='category'
                            onChange={handleSelect}
                        >
                            <Selectoption disabled selected>
                                {translateMe("Selectcatogry")}
                            </Selectoption>
                            {categoriesList?.map((cate) => {
                                return (
                                    <Selectoption key={cate.id} id={cate.id}>
                                        {Router.locale === "ar"
                                            ? cate.name_ar
                                            : cate.name}
                                    </Selectoption>
                                );
                            })}
                        </CatogryInput>
                    </ItemInput>
                    <ItemInput>
                        <LabelDiv>
                            <ItemLocation>{t("Location")}</ItemLocation>
                            {errorMsg.location && (
                                <ErrorLabel errorMsg={errorMsg}>
                                    {translateMe("LocationisRequired")}
                                </ErrorLabel>
                            )}
                        </LabelDiv>
                        <LocationInput name='location' onChange={handleSelect}>
                            <Locationoption disabled selected>
                                {translateMe("SelectLocation")}
                            </Locationoption>
                            {locationList?.map((location) => {
                                return (
                                    <Locationoption
                                        key={location.id}
                                        id={location.id}
                                        value={JSON.stringify(location)}
                                    >
                                        {Router.locale === "ar"
                                            ? location.name_ar
                                            : location.name}
                                    </Locationoption>
                                );
                            })}
                        </LocationInput>
                    </ItemInput>
                    <ItemInput>
                        <LabelDiv>
                            <ItemDescription>
                                {translateMe("Description")}
                            </ItemDescription>
                            {errorMsg.description && (
                                <ErrorLabel errorMsg={errorMsg}>
                                    {translateMe("DescriptionisRequired")}
                                </ErrorLabel>
                            )}
                        </LabelDiv>
                        <DescriptionInput
                            placeholder={translateMe(
                                "DescriptionItemplaceholder"
                            )}
                            rows='10'
                            cols='105'
                            name='description'
                            onChange={handleChange}
                        />
                    </ItemInput>
                    <ItemImageInput>
                        <Itemupload>
                            {translateMe("UploadPhotossection")}
                        </Itemupload>
                        <ImagesDiv>
                            {imagesList.length === 0 ? (
                                <UploadContainer>
                                    <ItemImage
                                        alt='thumbnail'
                                        src={thumbnail}
                                        width='0'
                                        height='0'
                                    />

                                    <LabelDiv>
                                        <Uploadspan>
                                            {translateMe("UploadPhotos")}
                                        </Uploadspan>
                                        {errorMsg.image && (
                                            <ErrorLabel errorMsg={errorMsg}>
                                                {translateMe(
                                                    "photosareRequired"
                                                )}
                                            </ErrorLabel>
                                        )}
                                    </LabelDiv>
                                    <UploadInput
                                        type='file'
                                        name='image'
                                        placeholder={translateMe(
                                            "UploadPhotosplaceholder"
                                        )}
                                        onChange={handleImageFile}
                                        multiple
                                    />
                                </UploadContainer>
                            ) : (
                                <>
                                    {imagesList.map((SingleImage) => {
                                        return (
                                            <DisplayContainer
                                                key={SingleImage.url}
                                                onClick={() =>
                                                    deleteImage(
                                                        SingleImage.name
                                                    )
                                                }
                                            >
                                                <DeleteIcon>
                                                    <Image
                                                        src='./addToItem/delete.svg'
                                                        alt='delete icon'
                                                        width={30}
                                                        height={30}
                                                    />
                                                </DeleteIcon>
                                                <ItemImage
                                                    alt='product Image'
                                                    src={SingleImage.url}
                                                    width='300'
                                                    height='300'
                                                    quality={100}
                                                />

                                                <LabelDiv>
                                                    <Uploadspan>
                                                        {SingleImage.name === 0
                                                            ? "photo name "
                                                            : SingleImage.name}
                                                    </Uploadspan>
                                                    {errorMsg.image && (
                                                        <ErrorLabel
                                                            errorMsg={errorMsg}
                                                        >
                                                            {translateMe(
                                                                "photosareRequired"
                                                            )}
                                                        </ErrorLabel>
                                                    )}
                                                </LabelDiv>
                                            </DisplayContainer>
                                        );
                                    })}
                                    <PlusDiv>
                                        +
                                        <UploadInput
                                            type='file'
                                            name='images'
                                            placeholder='Upload Photos'
                                            onChange={handleImageFile}
                                            multiple
                                        />
                                    </PlusDiv>
                                </>
                            )}
                        </ImagesDiv>
                    </ItemImageInput>
                    <Buttoncontainer>
                        <ConfirmButton
                            disabled={!imagesList}
                            imagesList={imagesList.length === 0 ? false : true}
                            type='submit'
                        >
                            <Confirmspan>{translateMe("Confirm")}</Confirmspan>
                        </ConfirmButton>
                        <Link href='/listOfItems'>
                            <CancelButton type='submit'>
                                <Cancelspan>{translateMe("Cancel")}</Cancelspan>
                            </CancelButton>
                        </Link>
                    </Buttoncontainer>
                </FormItem>
            </Formcontainer>
        </Pagecontainer>
    );
}

export default AddItem;
