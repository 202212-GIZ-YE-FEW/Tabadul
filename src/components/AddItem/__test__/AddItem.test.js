import renderer from "react-test-renderer";

import AddItem from "../Index";

it("renders correctly", () => {
    const tree = renderer.create(<AddItem />).toJSON();
    expect(tree).toMatchSnapshot();
});
