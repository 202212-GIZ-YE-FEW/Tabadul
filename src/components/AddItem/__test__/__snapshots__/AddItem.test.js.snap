// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="css-13um9b0"
  >
    <nav
      className="css-1x3zj1f"
    >
      <a
        className="css-1kc05pi"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="css-1q0bv5v"
          onClick={[Function]}
        >
          <img
            className="css-1kh5p4n"
            src="/images/Logo.svg"
          />
        </div>
      </a>
      <div
        className="css-cnauqz"
        onBlur={[Function]}
        onClick={[Function]}
      >
        <div
          className="css-w9k2l0"
        />
        <div
          className="css-w9k2l0"
        />
        <div
          className="css-w9k2l0"
        />
      </div>
      <div
        className="css-1t7ejvz"
      >
        <div
          className="css-11erx38"
        >
          <div
            className="css-cf8cyr"
          >
            <button
              className="css-1a7iigq"
              onBlur={[Function]}
              onClick={[Function]}
            >
              <img
                className="css-55gwj7"
                src="/images/users.svg"
              />
            </button>
          </div>
          <div
            className="css-cf8cyr"
          >
            <button
              className="css-1a7iigq"
              onBlur={[Function]}
              onClick={[Function]}
            >
              <img
                className="css-55gwj7"
                src="/images/flag.svg"
              />
              <img
                className="css-jcgd1c"
                src="/images/arrow.svg"
              />
            </button>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="css-u9y4yi"
  >
    <h2
      className="css-te3fgk"
    >
      Add Item 
    </h2>
    <div
      className="css-1ieito0"
    >
      <form
        className="css-0"
        method="post"
        onSubmit={[Function]}
      >
        <div
          className="css-vaakcf"
        >
          <div
            className="css-u2ayx9"
          >
            <label
              className="css-a1lghw"
            >
              Title
            </label>
          </div>
          <input
            className="css-175bpda"
            name="title"
            onChange={[Function]}
            placeholder="Placeholder"
            type="text"
          />
        </div>
        <div
          className="css-vaakcf"
        >
          <div
            className="css-u2ayx9"
          >
            <label
              className="css-1hrsy57"
            >
              Catogry
            </label>
          </div>
          <select
            className="css-1wke6u8"
            name="category"
            onChange={[Function]}
            placeholder="Select catogry"
            type="dropdown"
          >
            <option
              className="css-0"
              disabled={true}
              selected={true}
            >
              Select catogry
            </option>
          </select>
        </div>
        <div
          className="css-vaakcf"
        >
          <div
            className="css-u2ayx9"
          >
            <label
              className="css-1hrsy57"
            >
              Location
            </label>
          </div>
          <select
            className="css-1e1l5cm"
            name="location"
            onChange={[Function]}
          >
            <option
              className="css-0"
              disabled={true}
              selected={true}
            >
              Select Location
            </option>
          </select>
        </div>
        <div
          className="css-vaakcf"
        >
          <div
            className="css-u2ayx9"
          >
            <label
              className="css-1hrsy57"
            >
              Description
            </label>
          </div>
          <textarea
            className="css-11vrh9q"
            cols="105"
            name="description"
            onChange={[Function]}
            placeholder="Description Item"
            rows="10"
          />
        </div>
        <div
          className="css-17djkoe"
        >
          <label
            className="css-s7589y"
          >
            Upload Photos
          </label>
          <div
            className="css-138mc57"
          >
            <div
              className="css-nt1pnc"
            >
              <img
                alt="thumbnail"
                className="css-1g2jbla"
                data-nimg="1"
                decoding="async"
                height={40}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={40}
              />
              <div
                className="css-u2ayx9"
              >
                <span
                  className="css-1byxpog"
                >
                  Upload Photos
                </span>
              </div>
              <input
                className="css-1tzc5tf"
                multiple={true}
                name="image"
                onChange={[Function]}
                placeholder="Upload Photos"
                type="file"
              />
            </div>
          </div>
        </div>
        <div
          className="css-11bfjdc"
        >
          <button
            className="css-19qn6sk"
            disabled={false}
            type="submit"
          >
            <span
              className="css-1t0xqtl"
            >
              Confirm
            </span>
          </button>
          <a
            href="/listOfItems"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <button
              className="css-1hhxna"
              type="submit"
            >
              <span
                className="css-whghps"
              >
                Cancel
              </span>
            </button>
          </a>
        </div>
      </form>
    </div>
  </div>
  <div
    className="css-1yhlq7n"
  >
    <div
      className="css-kp9m8d"
    >
      <div
        className="css-xvkl1c"
      >
        <img
          alt=""
          className="css-n2b87z"
          data-nimg="1"
          decoding="async"
          height={50}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/images/footerlogo.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={50}
        />
      </div>
      <div
        className="css-xvkl1c"
      >
        <a
          className="css-1dbkwr8"
          href="/Aboutus"
        >
          About us
        </a>
        <a
          className="css-1dbkwr8"
          href="/Blogs"
        >
          Blogs
        </a>
        <a
          className="css-1dbkwr8"
          href="/Products"
        >
          Listed items
        </a>
      </div>
      <div
        className="css-xvkl1c"
      >
        <p>
          Contact Us
        </p>
        <div
          className="css-18icqaw"
        >
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/facebook .svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/instagram (2).svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/linkedin (2).svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
          <a
            className="css-1i6zk58"
          >
            <img
              alt=""
              className="css-3g96gi"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/images/share.svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </div>
      </div>
      <div
        className="css-xvkl1c"
      >
        <p>
          Address
        </p>
        <p>
          street 01,07
        </p>
        <p>
          Yemen, Sanaa
        </p>
      </div>
    </div>
  </div>
</div>
`;
